.pre-commit-config.yaml

```diff
@@ -103,6 +103,12 @@ repos:
     - id: black-nb
       files: '\.ipynb$'
 - repo: https://github.com/pdm-project/pdm
-  rev: 2.5.3
+  rev: 2.6.1
   hooks:
     - id: pdm-lock-check
+    - id: pdm-export
+      args: ['-o', 'requirements/docker.txt', '--without-hashes', '--prod']
+      files: ^pdm.lock$
+#    - id: pdm-export
+#      args: ['-o', 'requirements/test.txt', '--without-hashes', '--dev']
+#      files: ^pdm.lock$
```

Comment:
I was thinking that the export should actually run automatically after a `pdm lock` when a dev changes dependencies.
The workflow being:
1. Dev uses `pdm add` or `pdm update ...` to modify dependencies, each of those two commands trigger a lock / re-lock
2. Therefore, it should be possible to add a command into pdm's `post_lock` hook that runs the exports

The benefit of using pdm hooks being that they happen before pre-commit in the dev cycle and the extra step of re-adding the requirements file after pre-commit runs the export would be avoided.

The pdm-lock-check on the other hand is still a last barrier to catch any discrepancies between pyproject.toml and pdm.lock.

Reply:
Perfect! That approach makes a lot of sense to me. I'll look into `pdm-autoexport`.